core/src/main/java/io/micronaut/core/execution/ExecutionFlow.java

```diff
@@ -278,5 +277,36 @@ default CompletableFuture<T> toCompletableFuture() {
      */
     default void cancel() {
     }
+
+    /**
+     * The flat map function interface.
+     * @param <I> The input
+     * @param <O> The output
+     * @since 4.8
+     */
+    @FunctionalInterface
+    interface FlatMapFn<I, O> {
```

Comment:
I don't think we should add APIs just to avoid type pollution for reactor.

Also this would be a binary incompatible change, and I think ExecutionFlow is used in servlet at this point

Reply:
I have removed those changes